internal/host/hostcommands/instruction_manager.go

```diff
@@ -59,19 +61,24 @@ type InstructionConfig struct {
 	ServiceBaseURL           string            `envconfig:"SERVICE_BASE_URL"`
 	ServiceCACertPath        string            `envconfig:"SERVICE_CA_CERT_PATH" default:""`
 	ServiceIPs               string            `envconfig:"SERVICE_IPS" default:""`
+	ImageServiceBaseURL      string            `envconfig:"IMAGE_SERVICE_BASE_URL"`
+	ImageExpirationTime      time.Duration     `envconfig:"IMAGE_EXPIRATION_TIME" default:"4h"`
 	InstallerImage           string            `envconfig:"INSTALLER_IMAGE" default:"quay.io/edge-infrastructure/assisted-installer:latest"`
 	ControllerImage          string            `envconfig:"CONTROLLER_IMAGE" default:"quay.io/edge-infrastructure/assisted-installer-controller:latest"`
 	AgentImage               string            `envconfig:"AGENT_DOCKER_IMAGE" default:"quay.io/edge-infrastructure/assisted-installer-agent:latest"`
+	HostFSMountDir           string            `envconfig:"HOST_FS_MOUNT_DIR" default:"/host"`
 	SkipCertVerification     bool              `envconfig:"SKIP_CERT_VERIFICATION" default:"false"`
 	DiskCheckTimeout         time.Duration     `envconfig:"DISK_CHECK_TIMEOUT" default:"8m"`
 	ImageAvailabilityTimeout time.Duration     `envconfig:"IMAGE_AVAILABILITY_TIMEOUT" default:"16m"`
 	DisabledSteps            []models.StepType `envconfig:"DISABLED_STEPS" default:""`
-	ReleaseImageMirror       string
-	CheckClusterVersion      bool
+
+	ReleaseImageMirror  string
+	CheckClusterVersion bool
 }
 
 func NewInstructionManager(log logrus.FieldLogger, db *gorm.DB, hwValidator hardware.Validator, ocRelease oc.Release,
-	instructionConfig InstructionConfig, connectivityValidator connectivity.Validator, eventsHandler eventsapi.Handler, versionHandler versions.Handler) *InstructionManager {
+	instructionConfig InstructionConfig, connectivityValidator connectivity.Validator, eventsHandler eventsapi.Handler,
+	versionHandler versions.Handler, insecureIPXEURLs bool, authType auth.AuthType) *InstructionManager {
```

Comment:
authType can be in the config also I think.
No need to pass around another parameter.

Reply:
Removed it, thank you!